packages/react-native-codegen/src/parsers/__tests__/parsers-commons-test.js

```diff
@@ -778,3 +784,253 @@ describe('buildSchema', () => {
+    it("throws an 'IncorrectModuleRegistryCallArgumentTypeParserError' error if call expression arg is NOT a string literal", () => {
+      const flowAstWithNonStringLiteralCallExpressionArg = parser.getAst(
+        'export default TurboModuleRegistry.getEnforcing(Spec);',
+      );
+      const flowCallExpression =
+        flowAstWithNonStringLiteralCallExpressionArg.body[0].declaration;
+      const flowCallExpressionCalllee = flowCallExpression.callee.property.name;
+      const flowCallExpressionArg = flowCallExpression.arguments[0];
+
+      const expected = new IncorrectModuleRegistryCallArgumentTypeParserError(
+        hasteModuleName,
+        flowCallExpressionArg,
+        flowCallExpressionCalllee,
+        flowCallExpressionArg.type,
+      );
+
+      expect(() =>
+        parseModuleName(
+          hasteModuleName,
+          moduleSpecs[0],
+          flowAstWithNonStringLiteralCallExpressionArg,
+          parser,
+        ),
+      ).toThrow(expected);
+    });
+
+    it("doesn't throw an 'IncorrectModuleRegistryCallArgumentTypeParserError' error if call expression arg is a string literal", () => {
+      expect(() =>
+        parseModuleName(
+          hasteModuleName,
+          moduleSpecs[0],
+          flowAstWithOneCallExpression,
+          parser,
+        ),
+      ).not.toThrow(IncorrectModuleRegistryCallArgumentTypeParserError);
+    });
+  });
+
+  describe('throwIfUntypedModule', () => {
+    it("throws an 'UntypedModuleRegistryCallParserError' error if call expression is untyped", () => {
+      const flowAstWithUntypedCallExpression = parser.getAst(
+        "export default TurboModuleRegistry.getEnforcing('SampleTurboModule');",
+      );
+      const flowCallExpression =
+        flowAstWithUntypedCallExpression.body[0].declaration;
+      const flowCallExpressionCallee = flowCallExpression.callee.property.name;
+      const moduleName = flowCallExpression.arguments[0].value;
+      const expected = new UntypedModuleRegistryCallParserError(
+        hasteModuleName,
+        flowCallExpression,
+        flowCallExpressionCallee,
+        moduleName,
+      );
+
+      expect(() =>
+        parseModuleName(
+          hasteModuleName,
+          moduleSpecs[0],
+          flowAstWithUntypedCallExpression,
+          parser,
+        ),
+      ).toThrow(expected);
+    });
+
+    it("doesn't throw an 'UntypedModuleRegistryCallParserError' error if call expression is typed", () => {
+      expect(() =>
+        parseModuleName(
+          hasteModuleName,
+          moduleSpecs[0],
+          flowAstWithOneCallExpression,
+          parser,
+        ),
+      ).not.toThrow(UntypedModuleRegistryCallParserError);
+    });
+  });
+
+  // TODO: add/complete the tests for 'IncorrectModuleRegistryCallTypeParameterParserError' error; using 'parseModuleName' function.
+  //
+  // describe('throwIfIncorrectModuleRegistryCallTypeParameterParserError', () => {
```

Comment:
@cipolleschi 
I've tried various ways to test this error, but I couldn't find a way to make it work.
Since I'm not yet; too familiar with parsers' functionalities, syntax and use cases, I don't want to spend more time on this.
I've left the commented code below as a reference for the future.
If you or anyone know how to test this error properly, please feel free to add/complete the tests.
Thanks!

Reply:
I think that the only part missing is to test [this check](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/flow/parser.js#L72). So if the `TypeParameter` is not a `GenericTypeAnnotation`, it should throw.

Reply:
uhm... probably we can remove this check.

The problem here is that 
```
export default TurboModuleRegistry.getEnforcing('SampleTurboModule');
```
this result in an `UntypedModuleRegistryCallParserError`, because it is not generic.

The problem is that, to get a type for it, it **must** be a generic. I can't think of any other way for `getEnforcing` to be typed with a different type rather than generic. 🤔 

I don't think there is a way to create a valid Flow code which is typed with something that is not a generic. I looked at the [base test](https://github.com/facebook/react-native/blob/main/packages/react-native-codegen/src/parsers/__tests__/error-utils-test.js#L282-L304) for that throw and they are manually creating an AST that satisfy that requirement, but that AST probably can't be actually generated. 🤔